core/include/detray/tools/track.hpp

```diff
@@ -0,0 +1,177 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2020-2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+#include <climits>
+#include <optional>
+#include <tuple>
+
+// detray tools
+#include "detray/definitions/qualifiers.hpp"
+#include "detray/tools/track_parameterization.hpp"
+#include "detray/utils/indexing.hpp"
+
+namespace detray {
+
+constexpr scalar _overstep_tolerance = -1e-4;
+
+struct bound_track_parameters {
```

Comment:
Since the file is already called ```track```, why not actually use inheritance here? I know it is not done in traccc, but I think, it would make the code much more readable. Besides, I don't think the rays we use for the geometry scans should need a parametrization. They also don't have one in ACTS, for example.

Reply:
Hmm I don't think Inheritance for `track` is going to reduce the amount of codes at all.

Reply:
Perfect, thanks!